gossipd/routing.c

```diff
@@ -855,7 +964,8 @@ void handle_channel_update(struct routing_state *rstate, const u8 *update)
 	u8 *tag = tal_arr(tmpctx, u8, 0);
 	towire_short_channel_id(&tag, &short_channel_id);
 	towire_u16(&tag, direction);
-	queue_broadcast(rstate->broadcasts,
+	replace_broadcast(rstate->broadcasts,
+			&chan->msg_indexes[MSG_INDEX_CUPDATE_0 | direction],
```

Comment:
This suggests that `MSG_INDEX_CUPDATE_0` and `MSG_INDEX_CUPDATE_1` have a specific numeric relationship with each other (i.e. that `(MSG_INDEX_CUPDATE_0 | 1) == MSG_INDEX_CUPDATE_1`. maybe the `enum gossip_msg_index` declaration should note that changing the order is dangerous? Or maybe explicitly assign numbers?

Reply:
This is a good place for BUILD_ASSERT